### DISCUSSION TITLE
Static params in parent segment, but dynamic in child segment

### DISCUSSION THREAD

**Discussion by MidnightDesign** [vercel/next.js#73467] [App Router] [answered]
I have an internationalized site. The paths look something like this: `/[locale]/events/[slug]`. The locales supported by the site are specified in `'app/[locale]/layout.tsx` using `export const generateStaticParams = () => LOCALES.map((locale) => ({ locale }));`. Now I want this segment to 404 if a page is  requested in an unsupported locale, so I add `export const dynamicParams = false;`. The `[slug]` segment should still be dynamic though. But as soon as I add the `dynamicParams = false`, I get a 404 for any `/[locale]/events/[slug]` page.

In other words: For `/[locale]/events/[slug]`, how do I make `[locale]` static while keeping `[slug]` dynamic?

(Of course I could just manually call `notFound()` in the layout component if the locale is invalid, but then what are the route segment options there for?)

**Comment by icyJoseph** [comment, marked as answer]
I'd say go with `notFound`. The Route Segment Configs were put in place, IIRC, as a low level control, mostly for when `fetch` was not used, or when it was just not clear to the dev, what the implicit behavior would be.

The docs, for v15, as of today say:

<img width="663" alt="Screenshot 2024-12-03 at 18 56 45" src="https://github.com/user-attachments/assets/03120dc4-4d0d-4c69-9b9c-946fb6073d0b">

**Reply by MidnightDesign** [reply]
Oh wow. Totally didn't see that. Thanks!
